lib/shared/addon/components/form-match-expressions-k8s/component.js

```diff
@@ -97,9 +99,13 @@ export default Component.extend({
 
     operatorChanged(match, selected){
       if (selected.value === 'Exists' || selected.value === 'DoesNotExist'){
+        // set values to empty array to force input component to update
+        set(match, 'match.values', [])
+        delete match.match.values
+      } else if (!match.match.values) {
```

Comment:
If I've read the spec correctly an empty array is acceptable here but I updated this to remove the key instead to match rke2 behavior. This caused a bug in which setting the operator to `In List`, defining `values`,  switching to `Exists` or `DoesNotExist`, then back to `In List` would remove `values` from the match expression, but the input would still show the previously entered text - setting an empty array first forces the input to clear itself and better reflect the actual value of `values`

Reply:
Agree with the change